train.py

```diff
@@ -0,0 +1,302 @@
+    # base_model = tf.keras.models.load_model(model_dir)
+    # embedding = tf.keras.models.Model(
+    #     name="embedding_model",
+    #     inputs=base_model.inputs,
+    #     outputs=base_model.get_layer(name="dense_2").output,
+    # )
+    # embedding.trainable = False
+
+    # Prepare data
+    negative_files_str = [str(path) for path in negative_files]
+
+    # Split the list of files into train, validation, and test sets
+    train_negatives, test_negatives = train_test_split(
+        negative_files_str, test_size=0.1, random_state=42
+    )
+
+    # Print the number of files in each set
+    print(f"Number of train negative files: {len(train_negatives)}")
+    print(f"Number of test negative files: {len(test_negatives)}")
+
+    # TODO: Change this split with one that takes into account the origin
+    # of the recordings (the bot, synthetic, etc.)
+    positive_files_str = [str(path) for path in positive_files]
+    # Split the list of files into train, validation, and test sets
+    train_samples, test_samples = train_test_split(
+        positive_files_str, test_size=0.2, random_state=42
+    )
+    train_samples, valid_samples = train_test_split(
+        train_samples, test_size=0.3, random_state=42
+    )
+
+    # Print the number of files in each set
+    print(f"Number of train files: {len(train_samples)}")
+    print(f"Number of validation files: {len(valid_samples)}")
+    print(f"Number of test files: {len(test_samples)}")
+
+    background_noise = data_dir / "ours" / "noise"
+
+    # Prepare the model
+    model_settings = standard_microspeech_model_settings(3)
+
+    if toy:
+        num_epochs = 1
+        num_batches = 1
+        backprop_into_embedding = False
+    else:
+        num_epochs = 10
+        num_batches = 2
+        backprop_into_embedding = True
+
+    print("---Training model---")
+    _, model, _ = transfer_learn(
+        target="hey_kiwibot",
+        train_files=train_samples,
+        val_files=valid_samples,
+        unknown_files=train_negatives,
+        num_epochs=num_epochs,
+        num_batches=num_batches,
+        batch_size=64,
+        primary_lr=0.001,
+        backprop_into_embedding=backprop_into_embedding,
+        embedding_lr=0.0005,
+        model_settings=model_settings,
+        base_model_path=model_dir,
+        base_model_output="dense_2",
+        UNKNOWN_PERCENTAGE=50.0,
+        bg_datadir=background_noise,
+        csvlog_dest=str(model_dir.parent / "training_log.csv"),
+        continue_training=False,
+    )
+
+    model.save(model_dir.parent / "finetuned_model")
+
+    test_spectrograms = np.array([file2spec(model_settings, f) for f in test_samples])
+    # fetch softmax predictions from the finetuned model:
+    # (class 0: silence/background noise, class 1: unknown keyword, class 2: target)
+    predictions = model.predict(test_spectrograms)
+    categorical_predictions = np.argmax(predictions, axis=1)
```

Comment:
what? is that comment right? Do we only have 2 classes??

Reply:
AH shit, my bad, I copy it from the notebook without noticing

Reply:
Done